extensions/http-curl/protocols/RESTSender.cpp

```diff
@@ -181,7 +184,7 @@ C2Payload RESTSender::sendPayload(const std::string& url, const Direction direct
   const auto response_body_bytes = gsl::make_span(client.getResponseBody()).as_span<const std::byte>();
   logger_->log_trace("Received response: \"%s\"", [&] {return utils::StringUtils::escapeUnprintableBytes(response_body_bytes);});
   if (isOkay && !clientError && !serverError) {
-    if (payload.isRaw()) {
+    if (accepted_formats) {
```

Comment:
depending on the outgoing payload's `isRaw` to process the response seems to be a workaround, so they do not really correlate

Reply:
So why do we check if `accepted_formats` is set here (also maybe check if it is empty here as well)? If `accepted_formats` is not set the format is interpreted as `*/*` on the C2 server side, which means it can return either yaml or json, how does the processing of the response differ depending on the `accepted_formats`?

Reply:
as I understand the direct download feature (e.g. a GET to the flow url) is outside the scope of the c2 protocol, even though it was done through the `consumePayload` method, the newly added `fetch` allows us to be explicit about this, implementation-wise the `RESTSender` uses its `sendPayload` for both except if the `accepted_formats` is set (not `nullopt`) we do not try to parse the response as JSON expecting a command, but forward the response as-is